components/storage/blockdevice/COMPONENT_DATAFLASH/DataFlashBlockDevice.cpp

```diff
@@ -140,8 +140,7 @@ DataFlashBlockDevice::DataFlashBlockDevice(PinName mosi,
                                            PinName cs,
                                            int freq,
                                            PinName nwp)
-    :   _spi(mosi, miso, sclk),
-        _cs(cs, 1),
+    :   _spi(mosi, miso, sclk, cs),
```

Comment:
You're not actually using the `use_gpio_ssel` constructor that your description mentions, and that would make this work fully. This has to be `_spi(mosi, miso, sclk, cs, use_gpio_ssel)`, otherwise you just get the legacy behaviour that passes `cs` to the HAL, and doesn't let you manually manipulate it. (See documentation of constructor and `select`).

It may look like it's working on your K64F (?) test, but

1. some targets don't actually support CS in the HAL at all, so they won't get any chip selects. That's one reason nearly all  SPI-using code was manually manipulating it via GPIO in the first place. They know that HALs only support CS if it's built in to the SPI controller - HALs don't provide their own GPIO emulation otherwise.
1. if it does work, it means the HAL will be asserting CS during the write calls only, and the manual `select` won't be keeping it on. (But it will at least be locking).

Reply:
Fixed. Now using use_gpio_ssel.